src/version.cpp

```diff
@@ -63,6 +63,35 @@
 #include <sys/types.h>
 #endif
 
+#ifndef lengthof
+#define lengthof(x) sizeof(x)/sizeof(x[0])
```

Comment:
Do we really need to define these 2 definitions always? It seems you are only using them in this particular case:

```cpp
#if defined(WIN32) || defined(__CYGWIN__) || defined(__MINGW__)
```